### ISSUE TITLE
Mouse scrolling not working?

### ISSUE THREAD

**Issue by mattmeier612** [moonlight-stream/moonlight-chrome#426]
It looks like this has been an on and off issue for a while now but I can't scroll with a mouse wheel in the latest version.  I'm running Chrome 65.0.3325.146 in Win 10 with the 0.8.2 release.  Is there something I can do to fix this?

**Comment by Jorys-Paulin** [comment]
@mattmeier612 I just tested with my laptop running Chrome 64 and it worked. Updating to the latest version of Chrome 65, It suddenly doesn't work. Normal scrolling still works. Any ideas, @cgutman?

**Comment by PittRipply** [comment]
After latest Google Chrome update, mouse scrolling is disabled but all mouse buttons still active including the scroll wheel when used as a button. We will need to wait for Moonlight update for resolution.

**Comment by cgutman** [comment]
I have observed this issue on Chrome 65.0.3325.146 on Linux, but I can't reproduce it on macOS. It sounds like @mattmeier612 has confirmed it on Windows as well.

I can look, but I'm not optimistic that this will be something fixable on our end, since it seems to be (yet another) bug in NaCl/Chrome. Similar to the Win10 Fall Creators Update bug which required a workaround in Chrome, we may need to wait for the Chrome developers to fix this.

**Comment by makedir** [comment]
I have the same issue, randomly.

**Comment by PittRipply** [comment]
After installing the latest GeForce Experience Driver Upgrade, mouse scrolling issue seems to have been resolved.

**Comment by makedir** [comment]
Nope, nothing fixed, not working, just sometimes.

**Comment by caudal** [comment]
It seems working with some of games, but not with others.
Impossible with Far Cry 5 for exemple.

**Comment by laurenke** [comment]
I'm unsure how helpful this is, as I'm using a Chromebook instead of what everyone else is probably using, but I've been having this same scrolling issue with my wireless mouse, and just now resolved it by scrolling first on the trackpad (which worked), after which the mouse's scroll wheel also started working. I've tested this now three times and it's worked consistently.

**Comment by Jorys-Paulin** [comment]
@everyone Does update to Chrome 66 resolved the issue for some of you? I haven't already tried.

**Comment by ajrc0re** [comment]
@Jorys-Paulin Nope, just tested and scroll wheel still isnt working. Even just a inconvenient work around would be nice if something like that is possible, there are some games that REQUIRE the scroll wheel and are currently unplayable.

**Comment by Jorys-Paulin** [comment]
@ajrc0re So that bug isn't still resolved. Is there a workaround, @cgutman?

**Comment by nicholson101** [comment]
any movement on this?

**Comment by Jorys-Paulin** [comment]
@nicholson101 Let me test...

**Comment by Jorys-Paulin** [comment]
@nicholson101 Nope; I just tried and it still doesn't work. there's also that qwerty bug when typing azerty...

**Comment by nicholson101** [comment]
oh!? didn't know about that one! I knew the scrolling didn't work but wasn't sure if anything was being worked on to rectify

**Comment by Jorys-Paulin** [comment]
@nicholson101 That's, unfortunately, a bug on Nvidia's side, as far as I know. We can't really do anything apart from asking them to fix it.

**Comment by nicholson101** [comment]
OK. Hopefully it'll be fixed over time.

**Comment by SirHamlin** [comment]
I was having this problem as well.  It would work randomly.  I did solve it however by uninstalling and reinstalling chrome on the HOST machine.  Sounds strange, but it worked for me!

**Comment by Zibbibbo** [comment]
Temporary fix : just after the connection with moonlight (i use Remote Desktop Executable mtsc.exe), when the desktop of the host pc appears, if you use mouse scroll BEFORE to click and take the control of the remote pc the scroll wheel will work correctly.

If you fail this passage : press the "windows" button to go back on the client pc, then click on the moonlight icon in the windows bar and again, use scroll wheel BEFORE to click on the remote desktop.

Sorry for my english, I hope you understand what to do. Is really simple.

**Comment by SirHamlin** [comment]
Here is a fix that works for me.  When connecting to the gamestream PC pick the game you'd like and whiles moonlight is running through the connection process continually scroll on your mouse wheel.  So the steps would look something like this..

1. Start moonlight
2. Select the game that you would like to play from your gamestream list 
3. Click on the game
4. While moonlight is loading the game and preparing to stream scroll on your mouse wheel.  (scroll up and down over and over back and forth until you have connected to the game)

This works for me everytime and is easy to do.

**Comment by mechaflash** [comment]
+1 @SirHamlin Your steps works 100% of the time for me as well. Why it works, I imagine, is an engineering phenomenon :)

**Comment by jamest212** [comment]
Same happens for me, and the "fix" works as well. Asus C302CA running on beta channel.

**Comment by Jorys-Paulin** [comment]
@jamest212 If you're running the beta, you may be able to run the new QT version of Moonlight: https://github.com/moonlight-stream/moonlight-qt/releases. Maybe give it a try?

**Comment by mdshw5** [comment]
Just confirming that I started seeing this issue using the Moonlight extension on ChromeOS.

**Comment by cgutman** [comment]
Hmm, ChromeOS 89 was released recently. Perhaps there is a regression in scroll events for NaCl apps?

Google has a history of [breaking](https://github.com/moonlight-stream/moonlight-chrome/issues/503) Moonlight [again](https://github.com/moonlight-stream/moonlight-chrome/issues/492) and [again](https://bugs.chromium.org/p/chromium/issues/detail?id=601191) with Chrome updates. They have clearly stopped testing NaCl APIs sufficiently now that they're deprecated everywhere except ChromeOS.

**Comment by mdshw5** [comment]
Thanks for chiming in @cgutman. I wasn't aware that NaCl was being deprecated. Would you suggest using the QT client in a linux container for ChromeOS going forward? If so do you know of a reason to choose Snap vs. Flatpak vs. AppImage? This will be my first time installing a linux GUI app on ChromeOS (though I work in a shell all day) so any pointers would be appreciated. The fact that there are 3 ways to do much the same thing makes me think there's something I'm missing.

**Comment by mdshw5** [comment]
@cgutman Whatever you've done in 7da5e83864e4e8739a0e4df8766f3c5d5e7bbe76 has fixed mouse scrolling in the NaCl Moonlight app now! Thanks so much! My kids are ecstatic that they can game on their chromebooks again. We had set up the android and linux builds of moonlight but for whatever reason the NaCl version gives the best performance for us under ChromeOS.

**Comment by ProjectsEdward** [comment]
It appears to be broken again in the latest build of moonlight-chrome and the latest stable release of Chrome.

**Comment by mdshw5** [comment]
@ProjectsEdward I can confirm that mouse scrolling is also again broken on the latest stable ChromeOS :/

**Comment by cgutman** [comment]
Please upvote this issue I filed on the Chromium bug tracker: https://bugs.chromium.org/p/chromium/issues/detail?id=1214614

**Comment by mdshw5** [comment]
Done. Thanks for submitting a reproducible example!
